atd-toolbox/data_model/python/cris_import.py

```diff
@@ -0,0 +1,427 @@
+import csv
+from datetime import datetime
+import io
+import os
+import time
+from zoneinfo import ZoneInfo
+
+import requests
+
+from settings import COLUMN_ENDPOINT
+
+FILE_DIR = "cris_csvs"
+HASURA_ENDPOINT = "http://localhost:8084/v1/graphql"
+UPLOAD_BATCH_SIZE = 1000
+
+CRASH_UPSERT_MUTATION = """
+mutation UpsertCrashes($objects: [crashes_cris_insert_input!]!) {
+  insert_crashes_cris(
+    objects: $objects, 
+    on_conflict: {
+        constraint: crashes_cris_pkey,
+        update_columns: [$updateColumns]
+    }) {
+    affected_rows
+  }
+}
+"""
+
+UNIT_UPSERT_MUTATION = """
+mutation UpsertUnits($objects: [units_cris_insert_input!]!) {
+  insert_units_cris(
+    objects: $objects, 
+    on_conflict: {
+        constraint: unique_units_cris,
+        update_columns: [$updateColumns]
+    }) {
+    affected_rows
+  }
+}
+"""
+
+PERSON_UPSERT_MUTATION = """
+mutation UpsertPeople($objects: [people_cris_insert_input!]!) {
+  insert_people_cris(
+    objects: $objects, 
+    on_conflict: {
+        constraint: unique_people_cris,
+        update_columns: [$updateColumns]
+    }) {
+    affected_rows
+  }
+}"""
+
+CHARGES_DELETE_MUTATION = """
+mutation DeleteCharges($crash_ids: [Int!]!) {
+  delete_charges_cris(where: {crash_id: {_in: $crash_ids}}) {
+    affected_rows
+  }
+}
+"""
+
+CHARGES_INSERT_MUTATION = """
+mutation InsertCharges($objects: [charges_cris_insert_input!]!) {
+  insert_charges_cris(objects: $objects) {
+    affected_rows
+  }
+}
+"""
+
+mutations = {
+    "crashes": CRASH_UPSERT_MUTATION,
+    "units": UNIT_UPSERT_MUTATION,
+    "persons": PERSON_UPSERT_MUTATION,
+    "charges": CHARGES_INSERT_MUTATION,
```

Comment:
that's right. the issue we're dealing with is that we have no way to uniquely identify a charge record in the CRIS extract, so the only way we can guarantee we have the right data is to purge it. 

so we just have a `charges_cris` table for charges (no `_edits` or unified table), and a crash’s charges are deleted and re-inserted whenever a crash is updated. There is no change log for charges. 

This still feels a little clumsy to me. It might be slightly better to soft-delete these records? Maybe it's not worth overthinking. In any case, I've added this to the "Documentation" todos. thanks for flagging it.

Reply:
yep, also, the fact that charges have been broken for a long while without disruption to the VZ team's work also supports not worrying too much. the only hesitation that i have too is if this is the only exception to a soft delete that we have in the model and why that is.